app/actions/apps.ts

```diff
@@ -3,65 +3,79 @@
 
 import {Client4} from '@mm-redux/client';
 import {ActionFunc} from '@mm-redux/types/actions';
-import {AppCallResponse, AppForm, AppCallRequest} from '@mm-redux/types/apps';
+import {AppCallResponse, AppForm, AppCallRequest, AppCallType} from '@mm-redux/types/apps';
 import {AppCallTypes, AppCallResponseTypes} from '@mm-redux/constants/apps';
-import {sendEphemeralPost} from './views/post';
 import {handleGotoLocation} from '@mm-redux/actions/integrations';
 import {showModal} from './navigation';
 import {Theme} from '@mm-redux/types/preferences';
 import CompassIcon from '@components/compass_icon';
 import {getTheme} from '@mm-redux/selectors/entities/preferences';
 import EphemeralStore from '@store/ephemeral_store';
+import {makeCallErrorResponse} from '@utils/apps';
 
-export function doAppCall<Res=unknown>(call: AppCallRequest, intl: any): ActionFunc {
+export function doAppCall<Res=unknown>(call: AppCallRequest, type: AppCallType, intl: any): ActionFunc {
     return async (dispatch, getState) => {
-        const ephemeral = (text: string) => dispatch(sendEphemeralPost(text, call?.context.channel_id, call?.context.root_id));
         try {
-            const res = await Client4.executeAppCall(call) as AppCallResponse<Res>;
+            const res = await Client4.executeAppCall(call, type) as AppCallResponse<Res>;
             const responseType = res.type || AppCallResponseTypes.OK;
 
             switch (responseType) {
             case AppCallResponseTypes.OK:
-                if (res.markdown) {
-                    if (!call.context.channel_id) {
-                        return {data: res};
-                    }
-
-                    ephemeral(res.markdown);
-                }
                 return {data: res};
             case AppCallResponseTypes.ERROR:
-                return {data: res, error: {message: res.error}};
+                return {error: res};
             case AppCallResponseTypes.FORM: {
                 if (!res.form) {
-                    const errMsg = 'An error has occurred. Please contact the App developer. Details: Response type is `form`, but no form was included in response.';
-
-                    ephemeral(errMsg);
-                    return {data: res};
+                    const errMsg = intl.formatMessage({
+                        id: 'apps.error.responses.form.no_form',
+                        defaultMessage: 'Response type is `form`, but no form was included in response.',
```

Comment:
Just wondering if this errors are shown to the user somehow, if they are.. do they make sense for an end user? they seem more relevant to the plugin developer / sysadmin, I doubt an end user will understand the meaning of this type of errors.

Reply:
We feel the error should somehow be delivered specially for debugging. What do you think about a "default error" and a `console.log` statement here?

Reply:
console output is removed from the codebase for production builds, see https://reactnative.dev/docs/next/performance#using-consolelog-statements